src/learning/gmm.rs

```diff
@@ -157,11 +161,32 @@ impl GaussianMixtureModel {
             model_means: None,
             model_covars: None,
             log_lik: 0f64,
+            bic: 1f64,
             max_iters: 100,
             cov_option: CovOption::Full,
         }
     }
 
+    /// Constructs a Gaussian Mixture Model from components.
+    /// Useful for constructing a gmm from existing parts (IE json deserialization)
+    ///
+    pub fn deserialize(comp_count: usize,
```

Comment:
I'm wary of introducing these manually like this. Instead we should try to use rustc-serialize or serde. Should be tackled separately in #122